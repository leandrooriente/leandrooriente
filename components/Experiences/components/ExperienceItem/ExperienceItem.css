.wrapper {
  margin-bottom: 75px;
}

.wrapper:last-child {
  margin-bottom: 0;
}

.headline {
  line-height: 1.6;
  font-size: 28px;
}

.description {
  line-height: 1.4;
}

.description li {
  margin-bottom: 10px;
}

.stack {
  border-top: 1px solid var(--light-gray);
  margin-top: 20px;
  padding-top: 5px;
}

.badge {
  margin: 0 10px 10px 0;
}

.badge:last-child {
  margin: 0;
}