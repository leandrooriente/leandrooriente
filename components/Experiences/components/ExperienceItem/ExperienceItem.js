import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Badge from 'components/Badge';
import ExperienceShape from '../../propTypes/experience';
import css from './ExperienceItem.css';

const Experience = ({ experience }) => (
  <div className={css.wrapper}>
    <div className={css.headline}>
      <strong>{experience.companyName}</strong>
      <Fragment> | </Fragment>
      <span>{experience.startDate}</span> - <span>{experience.endDate}</span>
    </div>
    <div
      className={css.description}
      dangerouslySetInnerHTML={{__html: experience.description}} // eslint-disable-line
    />
    <div className={css.stack}>
      {experience.stack.map(tech => (
        <Badge className={css.badge} key={tech}>{tech}</Badge>
      ))}
    </div>
  </div>
);

Experience.propTypes = {
  experience: PropTypes.shape(ExperienceShape).isRequired,
};

export default Experience;
