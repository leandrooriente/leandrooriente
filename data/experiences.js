export default [{
  role: 'Front End Developer',
  startDate: 'September 2017',
  endDate: 'Current',
  companyName: 'Travix',
  companyUrl: 'https://www.travix.com',
  description: `
    <p><a href="https://www.travix.com/">Travix</a> is one of the leading global online travel agencies operating five brands in 39 countries.</p>

    <ul>
      <li>
        Frontend Team Leader responsible for flights results page and seat map widget. 
      </li>

      <li>
        Responsible for researching and implement a new project architecture for our next platform so we can scale it to more than 100 affiliates.
      </li>

      <li>
        Responsible for the migration of our current E2E implementation to a new one using Webdriver.io testing more than 40 sites with different features and configurations.
      </li>
    </ul>
  `,
  stack: [
    'React',
    'Jest',
    'Enzyme',
    'Express',
    'Webpack',
    'CSS Modules',
    'SASS',
    'Webdriver.io',
    'Frint.js',
    'SSR',
  ],
}, {
  role: 'Front End Developer',
  startDate: 'August 2016',
  endDate: 'October 2017',
  companyName: 'OLX Brasil',
  companyUrl: 'https://www.storiaimoveis.com.br',
  description: `
    <p><a href="https://www.storiaimoveis.com.br/">Storia Im√≥veis</a> is the real estate vertical product of OLX Brazil, the biggest tradding platfom in South America. The product was created to compete directly with the top 3 real estate products in Brazil.</p>

    <p>We had a small and agile team of 10 developers to build a Progressive Web App in React and Node.js</p>

    <p>We decided to use Redux and Redux Saga to deal with business logic and side effects and to keep our components as pure as possible.</p>

    <p>I was also responsible for all the properties search experience (listing and map) and for hiring most of ours developers.</p>
  `,
  stack: [
    'React',
    'Redux',
    'Redux-Saga',
    'Webpack',
    'Express',
    'PostCSS',
    'CSS Modules',
    'SSR',
    'AWS',
    'Google SDK',
  ],
}, {
  role: 'Front End Developer',
  startDate: 'November 2012',
  endDate: 'October 2016',
  companyName: 'VTEX',
  companyUrl: 'http://www.vtex.com',
  description: `  
    <p><a href="http://www.vtex.com/">VTEX</a> is the biggest E-commerce Platform on the South America and one of the top AWS cases in Brazil. They handle thousands of sales everyday with a great performance.</p>

    <p>I worked at VTEX focused on developing applications that allows the client to have full control of their shop.</p>
    
    <p>I was the front end lead developer of the authentication experience, payment gateway and notification center. </p>
    
    <p>I also improved the hiring process of VTEX and I was part of the team responsible for the company's technology events.</p>
  `,
  stack: [
    'React',
    'Node.js',
    'Javascript',
    'Angular',
    'SASS',
    'Less',
  ],
}, {
  role: 'Front End Developer',
  startDate: 'February 2012',
  endDate: 'October 2012',
  companyName: 'Peixe Urbano',
  companyUrl: 'https://www.peixeurbano.com.br',
  description: `
    <p><a href="https://www.peixeurbano.com.br/">Peixe Urbano</a> is the biggest local commerce product in Brazil.</p>

    <p>At Peixe Urbano I worked as a front end developer for the main platform and the local delivery product.</p>
  `,
  stack: [
    'Javascript',
    'CSS3',
    'SASS',
    'jQuery',
  ],
}];
