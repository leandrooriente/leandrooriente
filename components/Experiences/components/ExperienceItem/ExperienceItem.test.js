import React from 'react';
import { mount } from 'enzyme';
import experiences from 'data/experiences';
import ExperienceItem from './ExperienceItem';

describe('<ExperienceItem />', () => {
  it('should render <ExperienceItem /> component', () => {
    const wrapper = mount(
      <ExperienceItem experience={experiences[0]} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
