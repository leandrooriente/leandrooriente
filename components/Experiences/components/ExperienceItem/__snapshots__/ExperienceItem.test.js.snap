// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExperienceItem /> should render <ExperienceItem /> component 1`] = `
<Experience
  experience={
    Object {
      "companyName": "Travix",
      "companyUrl": "https://www.travix.com",
      "description": "
    <p>Travix is one of the leading global online travel agencies operating five brands in 35 countries, employing more than 500 people with over 45 different nationalities worldwide.</p>

    <ul>
      <li>
        Frontend Team Leader responsible for flights results and seat maps. 
      </li>

      <li>
        Responsible for the migration of our current E2E implementation to a new one using Webdriver.io testing more than 40 affiliates.
      </li>
    </ul>
  ",
      "endDate": "Current",
      "role": "Front End Developer",
      "stack": Array [
        "React",
        "Jest",
        "Enzyme",
        "Express",
        "Webpack",
        "CSS Modules",
        "SASS",
        "Webdriver.io",
        "Frint.js",
        "SSR",
      ],
      "startDate": "September 2017",
    }
  }
>
  <div
    className="wrapper"
  >
    <div
      className="headline"
    >
      <strong>
        Travix
      </strong>
       | 
      <span>
        September 2017
      </span>
       - 
      <span>
        Current
      </span>
    </div>
    <div
      className="description"
      dangerouslySetInnerHTML={
        Object {
          "__html": "
    <p>Travix is one of the leading global online travel agencies operating five brands in 35 countries, employing more than 500 people with over 45 different nationalities worldwide.</p>

    <ul>
      <li>
        Frontend Team Leader responsible for flights results and seat maps. 
      </li>

      <li>
        Responsible for the migration of our current E2E implementation to a new one using Webdriver.io testing more than 40 affiliates.
      </li>
    </ul>
  ",
        }
      }
    />
    <div
      className="stack"
    >
      <Badge
        className="badge"
        key="React"
      >
        <span
          className="badge badge"
        >
          React
        </span>
      </Badge>
      <Badge
        className="badge"
        key="Jest"
      >
        <span
          className="badge badge"
        >
          Jest
        </span>
      </Badge>
      <Badge
        className="badge"
        key="Enzyme"
      >
        <span
          className="badge badge"
        >
          Enzyme
        </span>
      </Badge>
      <Badge
        className="badge"
        key="Express"
      >
        <span
          className="badge badge"
        >
          Express
        </span>
      </Badge>
      <Badge
        className="badge"
        key="Webpack"
      >
        <span
          className="badge badge"
        >
          Webpack
        </span>
      </Badge>
      <Badge
        className="badge"
        key="CSS Modules"
      >
        <span
          className="badge badge"
        >
          CSS Modules
        </span>
      </Badge>
      <Badge
        className="badge"
        key="SASS"
      >
        <span
          className="badge badge"
        >
          SASS
        </span>
      </Badge>
      <Badge
        className="badge"
        key="Webdriver.io"
      >
        <span
          className="badge badge"
        >
          Webdriver.io
        </span>
      </Badge>
      <Badge
        className="badge"
        key="Frint.js"
      >
        <span
          className="badge badge"
        >
          Frint.js
        </span>
      </Badge>
      <Badge
        className="badge"
        key="SSR"
      >
        <span
          className="badge badge"
        >
          SSR
        </span>
      </Badge>
    </div>
  </div>
</Experience>
`;
